packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -2,59 +2,166 @@ import { FieldSectionContentType, FieldSectionType } from './fields';
 import { PickersTimezone } from './timezone';
 
 export interface AdapterFormats {
```

Comment:
I can extract in another PR if you want

What I did is

1. Improve the descriptions
2. Deprecate any format that we don't use
3. Deprecate any format that is always the same combination of smaller formats (`monthAndYear` + `monthAndDate`) because I would not like to create an `hoursAndMinutesAndSeconds` and an `hoursAndMinutesAndSecondsAndMilliseconds` etc...
4. Group in 4 families for readability
5. Describes where the format is being used internally